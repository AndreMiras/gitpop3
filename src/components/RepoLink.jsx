import React from "react";
import PropTypes from "prop-types";

const RepoLink = ({ nameWithOwner }) => (
  <a href={`https://github.com/${nameWithOwner}`}>{nameWithOwner}</a>
);
RepoLink.propTypes = {
  nameWithOwner: PropTypes.string.isRequired,
};

export default RepoLink;
