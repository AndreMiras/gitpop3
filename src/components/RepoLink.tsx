import { FunctionComponent } from "react";

type RepoLinkProps = {
  nameWithOwner: string;
};

const RepoLink: FunctionComponent<RepoLinkProps> = ({ nameWithOwner }) => (
  <a href={`https://github.com/${nameWithOwner}`}>{nameWithOwner}</a>
);

export default RepoLink;
