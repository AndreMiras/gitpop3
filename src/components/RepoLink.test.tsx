import { describe, it, test, expect, vi } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import RepoLink from "./RepoLink";

test("renders", () => {
  const tree = renderer
    .create(<RepoLink nameWithOwner="django/Django" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
