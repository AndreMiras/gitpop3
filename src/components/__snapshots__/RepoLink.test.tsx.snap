// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<a
  href="https://github.com/django/Django"
>
  django/Django
</a>
`;
